kani-compiler/src/kani_middle/attributes.rs

```diff
@@ -258,6 +473,27 @@ pub fn test_harness_name(tcx: TyCtxt, def_id: DefId) -> String {
     parse_str_value(&marker).unwrap()
 }
 
+/// Expect the contents of this attribute to be of the format #[attribute =
+/// "value"] and return the `"value"`
+pub fn expect_key_string_value(sess: &Session, attr: &Attribute) -> rustc_span::Symbol {
+    let span = attr.span;
+    let AttrArgs::Eq(_, it) = &attr.get_normal_item().args else {
+        sess.span_fatal(span, "Expected attribute of the form #[attr = \"value\"]")
```

Comment:
Please delay aborting the session while validating the attributes. You should return `Result<Symbol, ErrorGuaranteed>` instead.